text/0000-partial_types.md

```diff
@@ -0,0 +1,642 @@
+
+## Partial type access
+
+I propose to extend type system by adding type access after type name (for Product Types), like `(i16, &i32, &str).%full`.
+
+If we omit to write type access, it means `.%full` access. Symbol (`%` percent) mean percent or part of the whole thing (variable in our case).
+
+Access has similar names and meanings as lifetimes: `.%full`(full access, soft keyword), `.%_`(don't care how partial access is, soft keyword) and any other `.%a`(some "a" access). But in most cases we use detailed access `.%{}`.
+
+## Fields names and field-access
+
+Fields names inside detailed access:
+- named field-names for structs(and units)
+- unnamed numbered for tuples
+- `*` "all fields" quasi-field
+- `..` "rest of fields" quasi-field
+- `self` pseudo-field for unsupported types
+
+Filed-access can be in 4 values:
+- `<nothing>` or `%permit` - field is accessible,
+- `<omitting>` or %deny` - field is forbidden to use,
+- `%miss` (E) - it is uninitialized field, field is forbidden to use,
+- `%ignore` (F) - quasi-field access, which hides truthful field access.
+
+`%permit` filed-access is a default behavior: permit to read, to write, to borrow, to move. 
+
+_A word %permit is need **for explanations only**, just use default `<nothing>` access._
+
+`%deny` filed-access forbids to have an access to read, to write, to borrow, to move (like outside access to private field). It is a compile error if someone try to access.
+
+_A word %deny is need **for explanations mostly**, just use default `<omitted field>` access._ It can be used explicitly in tuple initialization and binding deconstructions.
+
+`%unfill` is a shortcut for `%miss _`.
+
+`%any` is a shortcut for `%ignore _`.
+
+## Detailed access and mutability
+
+Detailed field-access has next structure: `.%{%field-access1 field-name1, %field-access2 field-name2, ..}` 
+
+Detailed access is a set of unordered and unsorted field-names and their filed-accesses.
+
+First `%field-access1` field-access if it is omitted means `%permit`. First `%field-accessM` field-access for _first omitted_ filed-name means `%opposite`. 
+
+The rest of omitted `%field-accessN` field-accesses means `%same` (same as previous `%field-accessN-1`)
+
+| `%field-accessN-1` | `%same`   | `%opposite` |
+|--------------------|-----------|-------------|
+| `%permit`          | `%permit` | `%deny`     |
+| `%deny`            | `%deny`   | `%permit`   |
+| `%miss` (E)        | `%miss`   | `%permit`   |
+
+Examples:
+```rust
+struct Point {x: f64, y: f64, was_x: f64, was_y: f64}
+
+let pfull = Point {was_x: 4.0, was_y: 5.0};
+	// pfull : Point.%{was_x, was_y, %miss x, y}
+```
+
+Type access `.%full` is a shortcut for `.%{*}`.
+
+It is forbidden to have `.%{%deny *}` values and `.%{%deny *}` references (but who knows, maybe they are suitable uninhabited types, phantom types or proxy types).
+
+## Access-filter and Mutable Filter
+
+Rust consumer use same names for action and for type clarifications, so we follow this style.
```

Comment:
That may apply to some cases, like references and mutability, but it's not a requirement. For example, lifetimes are only allowed in type position, but not in expression position:

```rs
fn foo<'a>(value: &'a i32) {}

foo(&'static 42); // ERROR! Lifetime not allowed here
// instead write
foo(&42 as &'static i32);
```

To try and reduce the scope of this (very large) RFC, I'd only allow access filter in type position.

Reply:
Thanks, I'll try to find what is reducible!